File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    
    queue = deque((i,p) for i,p in enumerate(priorities))

    result = []
    while queue:
        current = queue.popleft()
        for q in queue:
            if q[1] > current[1]:
                queue.append(current)
                break
        else:
            result.append(current)
        
    for i, (n,m) in enumerate(result):
        if location == n:
            return i+1

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        
        Queue<Progress> queue = new LinkedList<>();
        
        for (int i=0; i<priorities.length; i++) {
            Progress progress = new Progress(i, priorities[i]);
            queue.offer(progress);
        }
        
        while (true) {
            Progress current = queue.poll();
            
            if (queue.stream().anyMatch(q -> q.priority > current.priority)) {
                queue.offer(current);
            } else {
                answer+=1;
                if (location == current.index) {
                    return answer;
                }
            }
        }
    }
    
    private class Progress {
        final int index;
        final int priority;
        
        public Progress(int index, int priority) {
            this.index = index;
            this.priority = priority;
        }
    }
}